src/test/java/org/xstefank/api/AuthenticationTestIT.java

```diff
@@ -0,0 +1,65 @@
+package org.xstefank.api;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.jboss.arquillian.junit.Arquillian;
+import org.xstefank.JaxRsApplication;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.net.URL;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class AuthenticationTestIT {
+
+    private static final String RETURN_STATUS_UNAUTHORIZED_PATH = "/test/fakeGithub/ReturnStatusUnauthorized";
+    private static final String RETURN_STATUS_INTERNAL_SERVER_ERROR_PATH = "/test/fakeGithub/ReturnStatusInternalServerError";
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        return ShrinkWrap
+                .create(WebArchive.class, "test.war")
+                .addClasses(FakeGitHub.class, JaxRsApplication.class);
+    }
+
+    @ArquillianResource
+    private URL deploymentUrl;
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testAuthenticationStatusUnauthorized() {
+        Assert.assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), getEndPointResponse(RETURN_STATUS_UNAUTHORIZED_PATH).getStatus());
+        testAuthentication(RETURN_STATUS_UNAUTHORIZED_PATH);
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testAuthenticationStatusInternalServerError() {
+        Assert.assertEquals(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), getEndPointResponse(RETURN_STATUS_INTERNAL_SERVER_ERROR_PATH).getStatus());
```

Comment:
I'm not sure if this is the right approach as you are basically making same rest call twice if I'm not mistaken. But ok, if you want to verify that the paths are right I will merge this

Reply:
Thanks :)